Core/Core/Inbox/Addressbook/View/AddressbookRoleView.swift

```diff
@@ -0,0 +1,168 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2023-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+
+struct AddressbookRoleView: View {
+    @ObservedObject private var viewModel: AddressbookRoleViewModel
+    @Environment(\.viewController) private var controller
+
+    init(model: AddressbookRoleViewModel) {
+        self.viewModel = model
+    }
+
+    public var body: some View {
+        ScrollView {
+            switch viewModel.state {
+            case .loading:
+                loadingIndicator
+            case .data:
+                if viewModel.isRolesViewVisible {
+                    rolesView
+                } else {
+                    peopleView
+                }
+            case .empty, .error:
+                Text("There was an error loading recipients.", bundle: .core)
```

Comment:
Font and colors should be set. We could add some padding in case the text doesn't fix the screen's width. I would vertically center align the text on the screen, now it sticks to the top. Let's also add "Pull to refresh to try again." after this message on a new line (\n).
![IMG_0193](https://github.com/instructure/canvas-ios/assets/72396990/27b6eaa3-418d-4b40-8e76-a9aaeba2dbd4)